mesonbuild/environment.py

```diff
@@ -332,7 +332,7 @@ def __init__(self, source_dir, build_dir, options):
             try:
                 self.coredata = coredata.load(self.get_build_dir())
                 self.first_invocation = False
-            except FileNotFoundError:
+            except (FileNotFoundError, EOFError):
```

Comment:
Shouldn't try to recover from the cmd line file in that case?

Reply:
I think we should. What's the difference between "my file was completely deleted" and "all of the content of my file was deleted?"

Reply:
FileNoteFoundError means we never configured the build dir, so it's a first configuration. Other errors means we configured but something is broken in the coredata file so we should read the cmd file and reconfigure from scratch.

Reply:
To be more specific, I think coredata.load() should raise a MesonException if it gets EOFError

Reply:
Okay, I  can move that to coredata.